src/simulator/ackermann_model.cpp

```diff
@@ -93,6 +103,38 @@ void AckermannModel::Step(const double &dt) {
   // Update the Pose
   pose_.translation += Eigen::Rotation2Df(pose_.angle) * d_vector;
   pose_.angle = AngleMod(pose_.angle + dtheta);
+  this->Transform();
+}
+
+void AckermannModel::Transform() {
+  Eigen::Rotation2Df R(math_util::AngleMod(pose_.angle));
+  Eigen::Vector2f T = pose_.translation;
+
+  for (size_t i=0; i < template_lines_.size(); i++) {
+    pose_lines_[i].p0 = R * (template_lines_[i].p0) + T;
+    pose_lines_[i].p1 = R * (template_lines_[i].p1) + T;
+  }
 }
 
+void AckermannModel::SetTemplateLines(const float r, const int num_segments){
+  // copied directly from human model. In future refractor the code to 
+  // enable inherentance might be more optimal. For now there is an excessive
+  // amount of unwanted code in HumanModel, which disables inherenting directly.
+
+  const float angle_increment = 2 * M_PI / num_segments;
+
+  Eigen::Vector2f v0(r, 0.);
```

Comment:
If you use something in a .cpp file (not in a header), you should add a using declaration for that specific function at the top, especially if it is used more than once.
ex: `using Eigen::Vectorf2f;`

This suggestion applies throughout your changes, not just here.

Reply:
I believe that the declarition is already defined in the top. Are you suggesting to change, for example, `Eigen::Vector2f` to `Vectorf2f`?